### PR TITLE
layout: Floor to 0 when computing intrinsic sizes with box-sizing:border-box

### PR DESCRIPTION
**Pull request by stshine**
Improves bitbucket.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16428)

### CODE REVIEW

File: components/layout/fragment.rs [hunk 1]
```diff
@@ -937,7 +937,8 @@ impl Fragment {
             }
 
             if self.style.get_position().box_sizing == box_sizing::T::border_box {
-                specified -= border_padding
+                specified -= border_padding;
+                specified = max(Au(0), specified);
```
Comment (asajeffrey) [inline components/layout/fragment.rs]: Nit: would this be easier to read as `specified = max(Au(0), specified - border_padding);`?

**Comment by asajeffrey** [comment]
Apart from the nit, lgtm! You can r=me.

**Comment by emilio** [comment]
Needs a test though.

**Comment by emilio** [comment]
@bors-servo r+

Thanks for adding the test @stshine! :)

**Pull request merged**